src/inmanta/data/__init__.py

```diff
@@ -5303,17 +5340,23 @@ async def insert(self, connection: Optional[asyncpg.connection.Connection] = Non
         self.make_hash()
         await super().insert(connection=connection)
         # TODO: On conflict or is not exists or just make every update an upsert?
+        resource_status = (
+            state.ResourceStatus.UNDEFINED if self.status is const.ResourceState.undefined else state.ResourceStatus.HAS_UPDATE
```

Comment:
These inserts into the resource persisted state table are the only ones where the server should actually write to it imo, all others should be through the scheduler.

Reply:
I agree.